LiteCore/Storage/SQLiteKeyStore.cc

```diff
@@ -409,26 +429,38 @@ namespace litecore {
         t.commit();
     }
 
-
-    void SQLiteKeyStore::deleteIndex(slice expression, IndexType type) {
-        alloc_slice expressionFleece;
-        const Array *params;
-        tie(expressionFleece, params) = parseIndexExpr(expression, type);
-        string indexName = SQLIndexName(params, type, true);
-
-        Transaction t(db());
+    void SQLiteKeyStore::_deleteIndex(slice name, IndexType type) {
```

Comment:
This method should sanity-check the `name` parameter too (probably use a common `validateIndexName` function.)